### PR TITLE
Modify mobile keyboard to be scrollable

### LINKED ISSUE #2872: Enhancement: Mobile Keyboard should allow users to scroll through keys.
This is for the mobile keyboard docked on the right side on mobile screens.

Currently, users need to click a button on the far right to cycle through the list of keys. 

https://github.com/source-academy/frontend/assets/112536096/3098563c-d0d7-4744-a081-c9468af59166

It would be easier if users can simply scroll horizontally to move through the list of keys, which would be more convenient when they need to use different keys when typing their programs.

https://github.com/source-academy/frontend/assets/112536096/4783fdd8-d5a5-4fd9-a3b6-422e0e70839a

### PR DESCRIPTION
**Pull request by Johnwz123**
### Description

Closes #2872 

Modified mobile keyboard key selection. Previously, users had to click a button to cycle through rows of different sets of keys. Now, users scroll horizontally through a single row containing every key.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to test

1. Access Source Academy either on a physical mobile device or use browser dev tools to shrink viewport dimensions to access mobile UI.
2. Go to the Source Academy Playground page and expand the draggable keyboard on the right side of the screen.
3. Attempt typing characters into the editor by scrolling to find and clicking the respective keys. Ensure key click is not registered while scrolling. (Note: Editor needs to be in focus (blinking cursor) to be able to type.)

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

File: src/commons/mobileWorkspace/MobileKeyboard.tsx [hunk 1]
```diff
@@ -31,53 +33,86 @@ const MobileKeyboard: React.FC<Props> = props => {
...
+
+    if (Math.abs(deltaX) > swipeThreshold || Math.abs(deltaY) > swipeThreshold) {
+      console.log('Swipe detected');
+    } else if (deltaTime < tapThresholdTime) {
+      console.log('Tap detected');
+      handleKeyRelease();
+    } else {
+      // Handle other cases or ignore
+      console.log('Other touch event detected');
+    }
```
Comment (RichDom2185) [inline src/commons/mobileWorkspace/MobileKeyboard.tsx:85]: We don't need the `console.log` statements right?

**Review by RichDom2185 — COMMENTED** [review]
Hi, thanks for working on this!

I only have one comment, otherwise looks good, thanks!

File: src/commons/mobileWorkspace/MobileKeyboard.tsx (hunk 1 above)
Comment (Johnwz123) [reply src/commons/mobileWorkspace/MobileKeyboard.tsx]: Nope. I've removed them.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**